docs/api-diffs/9.0/rc.1/diff.md

```diff
@@ -0,0 +1,1290 @@
++            public static IResourceBuilder<T> WithHttpHealthCheck<T>(this IResourceBuilder<T> builder, string? path = null, int? statusCode = default(int?), string? endpointName = null) where T : IResourceWithEndpoints;
+-            public static IResourceBuilder<T> WithHttpsEndpoint<T>(this IResourceBuilder<T> builder, int? port = default(int?), int? targetPort = default(int?), string? name = null, string? env = null, bool isProxied = true) where T : IResourceWithEndpoints;
++            public static IResourceBuilder<T> WithHttpsEndpoint<T>(this IResourceBuilder<T> builder, int? port = default(int?), int? targetPort = default(int?), [EndpointNameAttribute] string? name = null, string? env = null, bool isProxied = true) where T : IResourceWithEndpoints;
++            public static IResourceBuilder<T> WithHttpsHealthCheck<T>(this IResourceBuilder<T> builder, string? path = null, int? statusCode = default(int?), string? endpointName = null) where T : IResourceWithEndpoints;
+         }
+     }
+     namespace Aspire.Hosting.ApplicationModel {
+-        [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-        public class AfterEndpointsAllocatedEvent : IDistributedApplicationEvent
++        public class AfterEndpointsAllocatedEvent : IDistributedApplicationEvent
+-        [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-        public class AfterResourcesCreatedEvent : IDistributedApplicationEvent
++        public class AfterResourcesCreatedEvent : IDistributedApplicationEvent
+-        [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-        public class BeforeResourceStartedEvent : IDistributedApplicationEvent, IDistributedApplicationResourceEvent
++        public class BeforeResourceStartedEvent : IDistributedApplicationEvent, IDistributedApplicationResourceEvent
+-        [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-        public class BeforeStartEvent : IDistributedApplicationEvent
++        public class BeforeStartEvent : IDistributedApplicationEvent
++        public static class CommandResults {
++            public static ExecuteCommandResult Success();
++        }
++        public class ConnectionStringAvailableEvent : IDistributedApplicationEvent, IDistributedApplicationResourceEvent {
++            public ConnectionStringAvailableEvent(IResource resource, IServiceProvider services);
++            public IResource Resource { get; }
++            public IServiceProvider Services { get; }
++        }
+         public class ConnectionStringReference : IManifestExpressionProvider, IValueProvider, IValueWithReferences {
++            public string? ConnectionName { get; set; }
```

Comment:
@captainsafia - I'm not seeing this property used anywhere. Can it be removed?

Reply:
I added it, I think we should keep it. It properly captures all of the information from the call. It was missing before.

Reply:
Might be good to have a unit test for it...

Reply:
Yes, I believe we can remove it. This property predates our proposal to use `IAzureFunctionsResourceConfig` to inject environment variables into the Functions process. I can open a PR to remove it.

Reply:
> I added it, I think we should keep it. It properly captures all of the information from the call. It was missing before.

Looks like I hit send on my message before seeing yours. What do you mean by “properly capture all of the information from the call”?